marklogic-data-hub/src/main/java/com/marklogic/hub/dhs/installer/command/AbstractInstallerCommand.java

```diff
@@ -81,4 +84,21 @@ protected String getServerMajorVersion() {
             return "9";
         }
     }
+
+    protected ObjectNode canInstallDhs(String installedHubVersion, MarkLogicVersion mlVersion){
+        ObjectMapper mapper = new ObjectMapper();
+        ObjectNode node = mapper.createObjectNode();
+        if(Character.getNumericValue(installedHubVersion.charAt(0)) < 5){
+            node.put("canBeInstalled", false);
+            node.put("message", "DHF cannot be upgraded when the major version of the existing DHF instance is 4");
+        }
+        else if(mlVersion.getMajor().equals(9) || (mlVersion.getMajor().equals(10) && mlVersion.getMinor() < 300)){
```

Comment:
I think a better way of implementing this to future-proof it is to say:

- If major version is higher than 10, then can install
- If major version is 10, and minor version is 3 or higher, then can install 
- Else, cannot install